### DISCUSSION TITLE
Nova doesn't handle namespaces correctly

### DISCUSSION THREAD

**Discussion by digitalhuman** [laravel/nova-issues#3522] [Q&A]
Example:

Folder structure:

User resource in 'Admin' namespace:

Nova/Resources/Admin/Users/User.php

---

User resource in root namespace

Nova/Resources/User.php

---

Both URL's in Nova dashboard will go to '/nova/user'. Which should be  /nova/admin/users/user and /nova/user.

**Comment by crynobone** [comment]
This is expected behavior as resource will convert class name using `Str::plural(Str::kebab(class_basename(get_called_class())));`. As since all our `nova-api` routes expect single level route param for `resourceName` the following `nova/admin/users/user` isn't possible.

**Reply by digitalhuman** [reply]
No when using namespaces or folder structure the expected behavior would be that the URL's follow this structure. That it's not implemented as such is a different case.

**Reply by davidhemphill** [reply]
Semantics. It's expected behavior since the code was not written to consider namespaces, but the classname only.

**Reply by digitalhuman** [reply]
Could be a good improvement. More robust.
